pramen-py/src/pramen_py/metastore/reader.py

```diff
@@ -123,41 +145,44 @@ def get_latest_available_date(
             f"Getting latest available date for the table: {table_name} "
             f"until {until}"
         )
-        table = get_metastore_table(table_name, self.tables)
-
-        def before_until(date: datetime.date) -> bool:
-            return date <= until  # type: ignore
+        metastore_table = get_metastore_table(table_name, self.tables)
 
-        def extract_dates(file_name: str) -> datetime.date:
-            return convert_str_to_date(
-                file_name.split("=")[1],
-                fmt=table.info_date_settings.format,
+        if metastore_table.format == TableFormat.delta:
+            df_select = self._read_table(
+                metastore_table.format, metastore_table.path
+            ).select(f"{metastore_table.info_date_settings.column}")
+            dates_list = [data[0] for data in df_select.distinct().collect()]
+        else:
+            files = self.fs_utils.list_files(
+                metastore_table.path,
+                file_pattern=f"{metastore_table.info_date_settings.column}=*",
+            )
+            logger.debug(
+                f"The following files are in the {metastore_table.path}:\n"
+                f"{chr(10).join(files)}"
+            )
+            dates_list = self._extract_dates_from_files_name(
+                files, metastore_table.info_date_settings.format
             )
 
-        files = self.fs_utils.list_files(
-            table.path,
-            file_pattern=f"{table.info_date_settings.column}=*",
-        )
-        logger.debug(
-            f"The following files are in the {table.path}:\n"
-            f"{chr(10).join(files)}"
-        )
+        def before_until(date: datetime.date) -> bool:
+            return date <= until  # type: ignore
 
         try:
-            latest_date = max(
-                filter(
-                    before_until,
-                    map(
-                        extract_dates,
-                        files,
+            latest_date = max(filter(before_until, dates_list))
```

Comment:
I'd suggest, you can do a couple of checks before invoking `max`:
- If `dates_list` is empty throw an error saying that attempted to read the latest snapshot of an empty table.
- If `dates_list` is not empty, but after filtering with `filter(before_until, dates_list)` it IS empty, you can raise an error saying that no data available that is older than the specified 'until' date.

I suggest doing these checks before `try` so the error message reflects more regarding what is actually happening.

But you can keep the try..except block for other possible errors.